modules/core/src/main/scala/com.snowplowanalytics.iglu/client/validator/CustomMessageFormatter.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2014-2023 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.iglu.client
+package validator
+
+import java.text.MessageFormat
+import java.util.Properties
+import scala.jdk.CollectionConverters._
+
+/**
+ * Custom message formatter that maintains backward-compatible error messages
+ * regardless of json-schema-validator library version changes.
+ *
+ * Message templates are frozen at version 1.0.76 format to prevent breaking
+ * changes for downstream consumers (e.g., Enrich, DQD) that parse error messages.
+ *
+ * Templates are loaded from jsv-messages-1.0.76.properties, which is a copy of
+ * the original file from json-schema-validator 1.0.76.
+ */
+object CustomMessageFormatter {
+
+  /**
+   * Message templates from json-schema-validator 1.0.76.
+   * Format uses Java MessageFormat placeholders: {0}, {1}, {2}, etc.
+   *
+   * Template arguments:
+   * - {0} is always the JSON path (e.g., "$.field" or "$")
+   * - {1}, {2}, ... are validator-specific arguments
+   *
+   * Loaded from jsv-messages-1.0.76.properties (43 templates total)
+   */
+  private val messageTemplates: Map[String, String] = {
```

Comment:
How about making it a `Map[String, MessageFormat]`?  i.e. compile the templates one time only.